cs/src/Connections/RelayTunnelConnector.cs

```diff
@@ -121,7 +121,15 @@ public RelayTunnelConnector(IRelayClient relayClient)
                         case HttpStatusCode.TooManyRequests:
                             errorDescription = "Rate limit exceeded (429). Too many requests in a given amount of time.";
                             exception = new TunnelConnectionException(errorDescription, wse);
```

Comment:
It would be great to have its own exception type for the rate limit exception.
Ideally, with additional rate limit properties from the response headers. Unfortunately, there is no way to get response headers from web socket error in 6.0 (there is a way in 7.0 though)

Reply:
Created this issue to tracking doing that in the future https://github.com/microsoft/basis-planning/issues/541